table/sst_file_dumper.cc

```diff
@@ -206,23 +213,25 @@ uint64_t SstFileDumper::CalculateCompressedTableSize(
   for (iter->SeekToFirst(); iter->Valid(); iter->Next()) {
     table_builder->Add(iter->key(), iter->value());
   }
-  if (!iter->status().ok()) {
-    fputs(iter->status().ToString().c_str(), stderr);
-    exit(1);
+  s = iter->status();
+  if (!s.ok()) {
+    return s;
   }
-  Status s = table_builder->Finish();
+  s = table_builder->Finish();
   if (!s.ok()) {
-    fputs(s.ToString().c_str(), stderr);
-    exit(1);
+    return s;
   }
-  uint64_t size = table_builder->FileSize();
+  *compressed_table_size = table_builder->FileSize();
   assert(num_data_blocks != nullptr);
   *num_data_blocks = table_builder->GetTableProperties().num_data_blocks;
-  env->DeleteFile(testFileName);
-  return size;
+  s = env->DeleteFile(testFileName);
+  if (!s.ok()) {
+    return s;
+  }
+  return Status::OK();
```

Comment:
The shorter version (which I prefer but others might not) is `return env->DeleteFile(...);`  😄  That should be OK by the status check.